File: src/lib/paths.ts
import path from 'node:path';
import { mkdir } from 'node:fs/promises';

export class Paths {
  readonly root: string;

  constructor(rootDir: string = path.resolve(process.cwd(), '.codex-subagent')) {
    this.root = path.resolve(rootDir);
  }

  get stateDir(): string {
    return path.join(this.root, 'state');
  }

  get logsDir(): string {
    return path.join(this.root, 'logs');
  }

  get archiveRoot(): string {
    return path.join(this.root, 'archive');
  }

  get threadsFile(): string {
    return path.join(this.stateDir, 'threads.json');
  }

  logFile(threadId: string): string {
    return path.join(this.logsDir, `${threadId}.ndjson`);
  }

  archiveDir(threadId: string): string {
    return path.join(this.archiveRoot, threadId);
  }

  archivedLogFile(threadId: string): string {
    return path.join(this.archiveDir(threadId), 'log.ndjson');
  }

  async ensure(): Promise<void> {
    await Promise.all([
      mkdir(this.stateDir, { recursive: true }),
      mkdir(this.logsDir, { recursive: true }),
      mkdir(this.archiveRoot, { recursive: true }),
    ]);
  }

  resolve(...segments: string[]): string {
    return path.join(this.root, ...segments);
  }
}

File: src/codex_subagent/paths.py
from __future__ import annotations

from dataclasses import dataclass
from pathlib import Path


@dataclass
class Paths:
    root: Path

    def __post_init__(self) -> None:
        if not isinstance(self.root, Path):
            self.root = Path(self.root)

    @property
    def state_file(self) -> Path:
        return self.root / "state" / "threads.json"

    def ensure(self) -> None:
        self.root.mkdir(parents=True, exist_ok=True)
        (self.root / "state").mkdir(parents=True, exist_ok=True)
        (self.root / "logs").mkdir(parents=True, exist_ok=True)
